include/nbl/video/EApiType.h

```diff
@@ -0,0 +1,21 @@
+#ifndef __NBL_E_API_TYPE_H_INCLUDED__
+#define __NBL_E_API_TYPE_H_INCLUDED__
+
+#include "nbl/core/compile_config.h"
+#include <cstdint>
+
+namespace nbl {
+namespace video
+{
+
+enum E_API_TYPE : uint32_t
+{
+    EAT_OPENGL,
+    EAT_OPENGL_ES,
+    EAT_VULKAN
+};
+
+}
+}
```

Comment:
I hate having free floating enums like this, they're obviously only meant to be used to describe an `video::IAPIConnection` so they should live within that scope

Reply:
https://github.com/Devsh-Graphics-Programming/Nabla/pull/28#discussion_r577205326

Reply:
I said I'm fine with including it every time

Reply:
iapiconnection includes IPhysicalDevice, PhysicalDevice includes ILogicalDevice, ILogicalDevice includes god knows what else, it's jsut perfect way to have circular deps in includes. Why would you be fine with this

Reply:
> 
> 
> iapiconnection includes IPhysicalDevice, PhysicalDevice includes ILogicalDevice, ILogicalDevice includes god knows what else, it's jsut perfect way to have circular deps in includes. Why would you be fine with this

I guess the only reason that IAPIConnection needs to know about IPhysicalDevice is because `smart_refctd_ptr` needs to be instantiated with it?

In which case could we put `E_TYPE` in a base class or a `IAPIConnection` namespace ?

Reply:
seems like something that could be fixed with a declaration/definition convention

Reply:
ok it actually makes sense to be free floating